src/extension/wps/src/main/java/au/org/emii/aggregator/NetcdfAggregator.java

```diff
@@ -0,0 +1,331 @@
+                List<Dimension> variableDimensions = new ArrayList<Dimension>();
+
+                for (Dimension dimension: variable.getDimensions()) {
+                    for (Dimension fileDimension: fileDimensions) {
+                        if (fileDimension.getFullName().equals(dimension.getFullName())) {
+                            variableDimensions.add(fileDimension);
+                        }
+                    }
+                }
+
+                Variable fileVariable = writer.addVariable(GLOBAL, variable.getShortName(), variable.getDataType(), variableDimensions);
+
+                for (Attribute attribute: variable.getAttributes()) {
+                    writer.addVariableAttribute(fileVariable, attribute);
+                }
+            }
+
+            // Finished defining file contents
+
+            writer.create();
+
+            // Copy static data (coordinate axes, etc) to output file
+
+            for (NetcdfVariable variable: template.getVariables()) {
+                if (!variable.isUnlimited()) {
+                    Variable fileVariable = writer.findVariable(variable.getShortName());
+                    writer.write(fileVariable, variable.read());
+                }
+            }
+        } catch (IOException|InvalidRangeException e) {
+            throw new AggregationException("Could not create output file", e);
+        }
+    }
+
+    private void appendRecordVariables(NetcdfDatasetIF dataset) throws AggregationException {
+        try {
+            for (NetcdfVariable templateVariable: templateDataset.getVariables()) {
+                if (!templateVariable.isUnlimited()) {
+                    continue;
+                }
+
+                NetcdfVariable datasetVariable = dataset.findVariable(templateVariable.getShortName());
+                append(datasetVariable);
+            }
+
+            writer.flush();
+        } catch (IOException e) {
+            throw new AggregationException(e);
+        }
+    }
+
+    private void append(NetcdfVariable srcVariable) throws AggregationException {
+        try {
+            Variable destVariable = writer.findVariable(srcVariable.getShortName());
+            int[] shape = srcVariable.getShape();
+            int[] slice = srcVariable.getShape();
+            slice[0] = 1;
+            int[] srcOrigin = new int[shape.length];
+            int[] destShape = destVariable.getShape();
+            int[] destOrigin = new int[destShape.length];
+            destOrigin[0] = destShape[0];
+
+            for (int i = 0; i < shape[0]; i++) {
+                srcOrigin[0] = i;
+                Array data = srcVariable.read(srcOrigin, slice);
+                writer.write(destVariable, destOrigin, data);
+                destOrigin[0]++;
+            }
+        } catch (IOException |InvalidRangeException e) {
+            throw new AggregationException(e);
+        }
+    }
+
+    private Map<String, UnpackerOverrides> getUnpackerOverrides() {
```

Comment:
Unpackeroverrides are really to do with reading data from the source files - I don't think they should be sourced from the template dataset now.  I've made some changes here in later pull requests which I'd like to build on.  So will tackle separately to this PR/after other PR's are reviewed/merged.